xapian-applications/omega/postprocess.py

```diff
@@ -0,0 +1,118 @@
+                row.append(row[2])
+
+                # Convert Hitlist from string to list
+                hits = row[2]
+                hits = hits.strip().split(',')
+                row[2] = hits
+
+                # Convert Clicks from string to list
+                clicks = row[4]
+                clicks = clicks.strip().split(',')
+                row[4] = clicks
+
+                # Initialise Clicks list elements to zero
+                for i in range(len(row[4])):
+                    row[4][i] = str(0)
+
+                writer.writerow(row)
+except IOError:
+        print "Could not read files."
+
+try:
+    with open('log/clickstream.log', 'r+') as s:
+        with open(clicks_log, 'r') as f:
+            with open('log/final.log', 'w+') as w:
+                reader1 = csv.reader(s, delimiter = '\t')
+
+                # Store the contents of clicks_log in a list object
+                reader2 = list(csv.reader(f, delimiter = '\t'))
+
+                writer = csv.writer(w, delimiter = '\t', quoting = csv.QUOTE_MINIMAL)
+
+                # Add headers to final log file
+                writer.writerow(["QueryID", "Query", "Hits", "Offset", "Clicks"])
+
+                for row1 in reader1:
+                    # Convert column 3 and column 5 values to list type
+                    l2 = literal_eval(row1[2])
+                    l4 = literal_eval(row1[4])
+
+                    for row2 in reader2:
+                        docid = ""
+                        # Find matching query IDs in clickstream.log and clicks_log
+                        if row1[0] == row2[0]:
+                            docid = row2[1]
+
+                        index = -1
+                        # Find the index of clicked doc ID in "Hits" list
+                        for i in range(len(l2)):
+                            if docid == l2[i]:
+                                index = i
+                        # Update the count representing the number of clicks made
+                        # on clicked document from "Hits" list
+                        for j in range(len(l4)):
+                            if j == index:
+                                count = int(l4[index])
+                                count = count + 1
+                                l4[j] = str(count)
+                    # Update "Clicks" list with updated values
+                    row1[4] = l4
+                    # Write the contents of row iterator to the final log file
+                    writer.writerow(row1)
+except IOError:
+        print "Could not read files."
+
+# Generate Query file formatted as per Xapian Letor documentation.
+try:
+    with open('log/final.log', 'r') as s:
+        with open('log/query.txt', 'w+') as w:
+            reader = csv.DictReader(s, delimiter = '\t')
+            writer = csv.writer(w)
+
+            for row in reader:
+                writer.writerow([row['QueryID'],row['Query']])
+except IOError:
+    print "Could not read file."
```

Comment:
Errors should be printed to `stderr` not `stdout`, and the program should then `exit()` non-zero. Additionally, this exception could be a failure of write as well as read, so the error message isn't necessarily accurate. It might be easier for now to refactor to three functions for the three blocks, none with exception handling, and then wrap the whole lot in a single exception handler, printing out the exception message (which I believe will include the relevant filename if it's an opening error).

You should also:

 * write python3-compatible code (eg use `print` as a function, not an expression)
 * look into `argparse` so you can provide useful `-h` / `--help` output
 * only act as a program if `__name__ == "__main__"`, which is a common python pattern

Turning the bulk of this into functions would make it possible to write tests for them, as well.